### PR TITLE
Feature/tf2 filter

### PR DESCRIPTION
**Pull request by iluetkeb**
Library and node for TF filtering.

### CODE REVIEW

File: tf2_filter/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+For chains with multiple links, simple give all links in order as arguments.
+
+## Example
+
+You can run `tf2_filter_example_publisher.py` to give you an example /tf stream (with empty transforms...). It publishes two transforms. With the TF2Filter invocation above, only one will remain.
+
+# License
+
+tf2_filter is licensed under the [Apache Public License v2](LICENSE).
```
Comment (choeppler) [inline tf2_filter/README.md]: I guess the it is called either Apache License, Version 2 or Apache-2.0 license, the latter being the standard SPDX license identifier.
```suggestion
tf2_filter is open-sourced under the Apache-2.0 license. See the [LICENSE](LICENSE) file for details.
```

File: tf2_filter/docs/requirements.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+## Data
+
+  * As a user, I want the output to be TFMessage, so that I can simply remap /tf->/tf_filtered for the target node and everything else stays the same.
+  * As a CPU-limited user, I would like to receive a transform chain collapsed to a single transform so that I don't have to compute it myself.
+
+## Configuration
+
+  * As an integrator I want to specify to be able to specify the target chain in the configuration, so that the target nodes don't have to be modified.
```
Comment (ralph-lange) [inline tf2_filter/docs/requirements.md]: Super-minor: Comma after integrator - for consistency with subsequent point.

File: tf2_filter/include/tf2_filter/tf2_filter.hpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+typedef std::map<std::string, TargetSet> FilterMap;
+
+class TF2Filter
+{
+public:
+  explicit TF2Filter(const FilterMap & relevant_transforms)
+  : relevant_transforms_(relevant_transforms)
+  {
+  }
+  virtual ~TF2Filter()
```
Comment (ralph-lange) [inline tf2_filter/include/tf2_filter/tf2_filter.hpp]: virtual ~TF2Filter() = default;

File: tf2_filter/package.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format2.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="2">
+  <name>tf2_filter</name>
+  <version>0.0.0</version>
+  <description>TODO: Package description</description>
+  <maintainer email="ingo.luetkebohle@de.bosch.com">Luetkebohle Ingo (CR/AEX3)</maintainer>
+  <license>APL</license>
```
Comment (ralph-lange) [inline tf2_filter/package.xml]: Propose to use 'Apache-2.0' from https://spdx.org/licenses/

File: tf2_filter/package.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format2.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="2">
+  <name>tf2_filter</name>
+  <version>0.0.0</version>
+  <description>TODO: Package description</description>
+  <maintainer email="ingo.luetkebohle@de.bosch.com">Luetkebohle Ingo (CR/AEX3)</maintainer>
```
Comment (ralph-lange) [inline tf2_filter/package.xml]: Org unit may be omitted.

File: tf2_filter/scripts/tf2_filter_example_publisher.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+#! /usr/bin/env python3
```
Comment (ralph-lange) [inline tf2_filter/scripts/tf2_filter_example_publisher.py:1]: Add copyright header, please.

File: tf2_filter/src/tf2_filter_node_impl.cpp [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+: Node(name, ns, get_global_default_context(),
+    {}, initial_params, true, use_intraprocess_comms, true),
+  tf_in_(this->create_subscription<tf2_msgs::msg::TFMessage>(
+      "/tf", std::bind(&TF2FilterNode::tf_cb, this, _1))),
+  tf_filter_out_(this->create_publisher<tf2_msgs::msg::TFMessage>(
+      "/tf_filtered")),
+  filter_(filtermap_from_param(this->get_parameter("frames")))
+{
+}
+TF2FilterNode::~TF2FilterNode()
```
Comment (ralph-lange) [inline tf2_filter/src/tf2_filter_node_impl.cpp]: You may use = default here.

File: tf2_filter/docs/requirements.md (hunk 2 above)
Comment (ralph-lange) [inline tf2_filter/docs/requirements.md]: Super-minor: Comma after integrator - for consistency with subsequent points.

File: tf2_filter/include/tf2_filter/tf2_filter.hpp (hunk 3 above)
Comment (ralph-lange) [inline tf2_filter/include/tf2_filter/tf2_filter.hpp:37]: virtual ~TF2Filter() = default;

**Review by ralph-lange — CHANGES_REQUESTED** [review]
Very good (and fast) work!

File: tf2_filter/package.xml (hunk 4 above)
Comment (choeppler) [reply tf2_filter/package.xml]: +1

Comment (iluetkeb) [reply tf2_filter/package.xml]: Done.

File: tf2_filter/package.xml (hunk 5 above)
Comment (iluetkeb) [reply tf2_filter/package.xml]: Removed.

File: tf2_filter/src/tf2_filter_node_impl.cpp (hunk 7 above)
Comment (iluetkeb) [reply tf2_filter/src/tf2_filter_node_impl.cpp]: Applied.

File: tf2_filter/include/tf2_filter/tf2_filter.hpp (hunk 3 above)
Comment (iluetkeb) [reply tf2_filter/include/tf2_filter/tf2_filter.hpp:37]: Applied.

File: tf2_filter/docs/requirements.md (hunk 2 above)
Comment (iluetkeb) [reply tf2_filter/docs/requirements.md]: Inserted.

**Comment by iluetkeb** [comment]
All comments resolved, merging.

Signed-off-by: Luetkebohle Ingo <ingo.luetkebohle@de.bosch.com>

**Pull request merged**